src/explorer/AzureAccountTreeItem.ts

```diff
@@ -54,4 +42,42 @@ export class AzureAccountTreeItem extends AzureAccountTreeItemBase {
         }
         return super.compareChildrenImpl(item1, item2);
     }
+
+    public async pickTreeItemImpl(expectedContextValues: (string | RegExp)[]): Promise<AzExtTreeItem | undefined> {
+        const subscription: string = localize('subscription', 'subscription');
+        const subscriptionOrTrialApp: string = localize('subscriptionOrTrialApp', 'subscription or trial app');
+        if (expectedContextValues.includes(TrialAppTreeItem.contextValue) && this.trialAppNode) {
+            this.childTypeLabel = subscriptionOrTrialApp;
+        } else {
+            this.childTypeLabel = subscription;
+        }
+
+        return super.pickTreeItemImpl(expectedContextValues);
+    }
+
+    private async loadTrialAppNode(): Promise<AzExtTreeItem | undefined> {
+        if (this.trialAppNode) {
+            await this.trialAppNode.refresh();
+            return this.trialAppNode;
+        } else {
+            const loginSession: string | undefined = ext.context.globalState.get(TrialAppLoginSession);
+            if (loginSession) {
+                const ti: AzExtTreeItem[] = await this.createTreeItemsWithErrorHandling(
+                    [loginSession],
+                    'trialAppInvalid',
+                    async (source: string): Promise<AzExtTreeItem> => {
+                        return await TrialAppTreeItem.createTrialAppTreeItem(this, source);
+                    },
+                    (_source: unknown): string => {
+                        return 'Trial App';
+                    });
+                if (ti.length > 0) {
+                    const treeItem: AzExtTreeItem = ti[0];
+                    this.trialAppNode = treeItem instanceof TrialAppTreeItem ? treeItem : this.trialAppNode;
```

Comment:
I don't understand this logic.  Wouldn't you want this to be undefined it we couldn't create a treeItem?